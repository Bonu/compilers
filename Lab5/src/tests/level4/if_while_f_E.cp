// if_whileE: Check statement types: new array, assignments, and array use.
class Main {

  void main() {

    int A[5];     int B[5];
    int i;        int j;
    bool p;       bool q;
    float f;

    A = B;
    i = p;
    while ( i) {
      A[i] = B[i]+0.5;
	{
	   int k;
	   bool f;
	   j = 0;
           while ((j < 10) && i) {
               A[i] = 0.3;
	       j = j + 1;
	   }
           f = k%(k*2);
           a = a + 1;
	}
	   if (p == q) p = q;
	   if (i+j >= A[j*.2])
	       i = j+10.1;
	   else {
	       p = p && f;
	       i = i%p;
           }
    }

	while ((i > 0) && (j == i)) {
		i = (i>0) || (j>0);
	}

  }
  int m(int b[]) {
	char ch[5];
	float y = x[0];
	char ch0 = z;
  }

  float x = 0.0;

}
