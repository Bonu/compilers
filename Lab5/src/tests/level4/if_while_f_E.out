line 12: the target ` i ' and source have incompatible types
          -- expects `int' but found `bool'
line 13: a conditional expression must be a boolean type
line 14: the target ` A [ i ] ' and source have incompatible types
          -- expects `int' but found `float'
line 19: invalid argument 2 to operator `&&'
          -- expects a boolean argument but found int
line 20: the target ` A [ i ] ' and source have incompatible types
          -- expects `int' but found `float'
line 23: the target ` f ' and source have incompatible types
          -- expects `bool' but found `int'
line 24: unable to resolve name -- a
line 24: unable to resolve name -- a
line 27: invalid array index -- requires int or char type 
           but found ` j * .2 ' which has type float
line 28: the target ` i ' and source have incompatible types
          -- expects `int' but found `float'
line 30: invalid argument 2 to operator `&&'
          -- expects a boolean argument but found float
line 31: invalid argument 2 ` p ' for arithmetic operator `%'
          -- expects a numeric argument but found bool
line 36: the target ` i ' and source have incompatible types
          -- expects `int' but found `bool'
line 42: ` x ' is not an array variable
line 43: unable to resolve name -- z
15 errors were detected
